ambry-store/src/main/java/com.github.ambry.store/PersistentIndex.java

```diff
@@ -51,11 +51,12 @@
   /**
    * Represents the different types of index entries.
    */
-  private enum IndexEntryType {
+  enum IndexEntryType {
     ANY, PUT, DELETE,
   }
 
-  static final short VERSION = 0;
+  static final short VERSION_0 = 0;
+  static final short VERSION_1 = 1;
```

Comment:
Why `PersistentIndex`, `IndexSegment`, and `IndexValue` all share the same version number? To me, it makes more sense to just change the version number for `IndexValue`, because this is the main body that has been changed to include more fields. This change, on the other side, should be transparent to `IndexSegment` and `PersistentIndex`. I prefer the version number kept only for local scope, which will be easier to make any further changes. For example, when changing service id field from `short` to `int`, it would not impact the entire index.

I do see the additional fields introduced to `IndexSegment`, which could be addressed by changing the version number of `IndexSegment` itself, which is irrelevant to the version number of `IndexValue`.

In short, each level should maintain its own version number. Changes at one level would not impact the other two. Please let me know if you need more discussions.